poc-07.py

```diff
@@ -68,7 +89,16 @@ def search(self):
                 # Missing in the other side.
                 changedMessages.append(message)
             else:
-                pass # TODO: compare: did message change?
+                for stateMessage in stateMessages:
+                    if message.uid == stateMessage.uid:
+                        if not message.identical(stateMessage):
+                            changedMessages.append(message)
+
+        for stateMessage in stateMessages:
+            if stateMessage not in messages:
+                # TODO: mark message as destroyed from real repository.
+                changedMessages.append(message)
+
         return changedMessages
 
```

Comment:
That's all I could come with for now. I wonder we could improve this by improving the Messages (collection) class.